src/main/java/uk/gov/hmcts/ccd/domain/model/casedataaccesscontrol/mapper/RoleAssignmentToAccessProfileMapperImpl.java

```diff
@@ -0,0 +1,89 @@
+import java.util.List;
+import java.util.stream.Collectors;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.AccessProfile;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.GrantType;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignment;
+import uk.gov.hmcts.ccd.domain.model.casedataaccesscontrol.RoleAssignmentFilteringResult;
+import uk.gov.hmcts.ccd.domain.model.definition.CaseTypeDefinition;
+import uk.gov.hmcts.ccd.domain.model.definition.RoleToAccessProfileDefinition;
+
+@Component
+public class RoleAssignmentToAccessProfileMapperImpl implements RoleAssignmentToAccessProfileMapper {
+
+    @Override
+    public List<AccessProfile> toAccessProfiles(RoleAssignmentFilteringResult filteringResult,
+                                                CaseTypeDefinition caseTypeDefinition) {
+        List<RoleAssignment> roleAssignments = extractRoleAssignments(filteringResult);
+
+        if (hasGrantTypeExcluded(roleAssignments)) {
+            roleAssignments = filterRoleAssignments(roleAssignments);
+        }
+
+        List<AccessProfile> accessProfiles = new ArrayList<>();
+        for (RoleAssignment roleAssignment : roleAssignments) {
+            for (RoleToAccessProfileDefinition roleToAccessProfile : caseTypeDefinition.getRoleToAccessProfiles()) {
+                if (validRoleAssignments(roleToAccessProfile, roleAssignment)) {
+
+                    List<String> authorisationsList = Arrays.asList(roleToAccessProfile.getAuthorisation().split(","));
+
+                    List<String> roleAssignmentAuthorisation = roleAssignment.getAuthorisations();
+
+                    if (roleAssignmentAuthorisation != null && roleAssignmentAuthorisation.size() > 0) {
+                        for (String authorisation : authorisationsList) {
+                            if (roleAssignmentAuthorisation.contains(authorisation)) {
+                                accessProfiles.add(createAccessProfile(roleAssignment, roleToAccessProfile));
+                            }
+                        }
+                    }
+                }
+            }
+        }
+        return accessProfiles;
+    }
+
+    private boolean validRoleAssignments(RoleToAccessProfileDefinition roleToAccessProfileDefinition,
+                                         RoleAssignment roleAssignment) {
+        return !roleToAccessProfileDefinition.getDisabled()
+            && roleAssignment.getRoleName().equals(roleToAccessProfileDefinition.getRoleName())
+            && !StringUtils.isEmpty(roleToAccessProfileDefinition.getAuthorisation());
+    }
+
+    private List<RoleAssignment> extractRoleAssignments(RoleAssignmentFilteringResult filteringResult) {
+        return filteringResult.getRoleAssignmentRoleMatchingResults()
+            .stream()
+            .map(pair -> pair.getLeft())
+            .collect(Collectors.toList());
+    }
```

Comment:
move this to `RoleAssignmentFilteringResult`, that's the whole point of adding these classes, so we can put all sorts of utility methods to them